catalog/dags/data_refresh/distributed_reindex.py

```diff
@@ -0,0 +1,425 @@
+        environment=environment,
+        target_environment=target_environment,
+        launch_template_version=launch_template_version,
+        aws_conn_id=aws_conn_id,
+    )
+
+    # Wait for the worker to finish initializing
+    await_worker = wait_for_worker.override(
+        poke_interval=data_refresh_config.reindex_poke_interval,
+        timeout=data_refresh_config.indexer_worker_timeout.total_seconds(),
+    )(environment=environment, instance_id=instance_id, aws_conn_id=aws_conn_id)
+
+    instance_ip_address = get_instance_ip_address(
+        environment=environment, instance_id=instance_id, aws_conn_id=aws_conn_id
+    )
+
+    worker_conn = create_connection(
+        instance_id=instance_id,
+        server=instance_ip_address,
+        environment=environment,
+        target_environment=target_environment,
+    )
+
+    trigger_reindexing_task = TempConnectionHTTPOperator(
+        task_id="trigger_reindexing_task",
+        http_conn_id=worker_conn,
+        endpoint="task",
+        data={
+            "model_name": data_refresh_config.media_type,
+            "table_name": data_refresh_config.media_type,
+            "target_index": target_index,
+            "start_id": start_id,
+            "end_id": end_id,
+        },
+        response_check=lambda response: response.status_code == 202,
+        response_filter=response_filter_status_check_endpoint,
+    )
+
+    wait_for_reindexing_task = TempConnectionHTTPSensor(
+        task_id="wait_for_reindexing_task",
+        http_conn_id=worker_conn,
+        endpoint=trigger_reindexing_task.output,
+        method="GET",
+        response_check=response_check_wait_for_completion,
+        mode="reschedule",
+        poke_interval=data_refresh_config.reindex_poke_interval,
+        timeout=data_refresh_config.indexer_worker_timeout,
+    )
+
+    terminate_instance = terminate_indexer_worker.override(
+        # Terminate the instance even if there is an upstream failure
+        trigger_rule=TriggerRule.ALL_DONE
+    )(
+        environment=environment,
+        instance_id=instance_id,
+    )
+
+    drop_conn = drop_connection(worker_conn=worker_conn)
+
+    instance_id >> [await_worker, instance_ip_address]
+    worker_conn >> trigger_reindexing_task >> wait_for_reindexing_task
+    wait_for_reindexing_task >> [terminate_instance, drop_conn]
+
+
+@task_group(
+    group_id="run_distributed_reindex",
+)
+def perform_distributed_reindex(
+    environment: str,
+    target_environment: Environment,
+    target_index: str,
+    data_refresh_config: DataRefreshConfig,
+    aws_conn_id: str = AWS_CONN_ID,
+):
+    """Perform the distributed reindex on a fleet of remote indexer workers."""
+    estimated_record_count = PGExecuteQueryOperator(
+        task_id="get_estimated_record_count",
+        conn_id=OPENLEDGER_API_CONN_ID,
+        sql=dedent(
+            f"""
+            SELECT id FROM {data_refresh_config.media_type}
+            ORDER BY id DESC LIMIT 1;
```

Comment:
This will need to be the temporary table too I think. I'd also be worried about this not being super performant, but I guess we can see how long it takes in production before trying to do something hackier!